src/main/cpp/feedback/DragonLeds.cpp

```diff
@@ -53,125 +67,160 @@ void DragonLeds::setOff()
-    }
+    frc::LEDPattern base = frc::LEDPattern::Solid(colar);
+    frc::LEDPattern pattern = base.Blink(cycleTime);
+
+    pattern.ApplyTo(m_ledBuffer);
 }
 
-void DragonLeds::setDiagnosticLED(int id, std::array<int, 3> color)
+void DragonLeds::SetAlternatingColorBlinkingPattern(frc::Color c1, frc::Color c2)
 {
-    if (id <= m_numberofDiagnosticLEDs)
+    if (m_ledBuffer.size() > 0)
     {
-        m_ledBuffer[id].SetRGB(color[0], color[1], color[2]);
+        if (m_timer > 2 * m_blinkPatternPeriod)
+            m_timer = 0;
+
+        int blinkState = (m_timer / m_blinkPatternPeriod) % 2;
+
+        if (blinkState == 0)
+            SetAlternatingColor(c1, c2);
+        else
+            SetAlternatingColor(c2, c1);
+
+        m_timer++;
     }
 }
 
-void DragonLeds::setBufferAllLEDsBlack()
+void DragonLeds::SetChaserPattern(frc::Color c)
 {
-    setBufferAllLEDsColor(getColorValues(BLACK));
+    if (m_ledBuffer.size() > 0)
+    {
+        m_loopThroughIndividualLEDs += m_loopThroughIndividualLEDs < static_cast<int>(m_ledBuffer.size()) - 1 ? 1 : -m_loopThroughIndividualLEDs;
+        if (!m_switchColor)
+        {
+            m_lastColor = m_lastColor == c ? frc::Color::kBlack : c;
+        }
+        m_switchColor = m_loopThroughIndividualLEDs != static_cast<int>(m_ledBuffer.size()) - 1;
+
+        SetSpecificLED(m_loopThroughIndividualLEDs, m_lastColor);
+    }
 }
 
-void DragonLeds::setBufferAllLEDsColorBrightness(Colors c, int brightness)
+void DragonLeds::SetClosingInChaserPattern(frc::Color c)
 {
-    std::array<int, 3U> hsvColor = getColorHSV(c);
-    hsvColor[2] = brightness;
-    for (unsigned int i = m_numberofDiagnosticLEDs; i < m_ledBuffer.size(); i++)
+    if (m_ledBuffer.size() > 0)
     {
-        m_ledBuffer[i].SetHSV(hsvColor[0], hsvColor[1], hsvColor[2]);
+        if (m_timer == 7)
+        {
+            int halfLength = (m_ledBuffer.size() - 1) / 2;
+            m_loopThroughIndividualLEDs += m_loopThroughIndividualLEDs < halfLength ? 1 : -m_loopThroughIndividualLEDs;
+            int loopout = (m_ledBuffer.size() - 1) - m_loopThroughIndividualLEDs;
+            auto color = m_colorLoop >= 0 ? c : frc::Color::kBlack;
+            m_colorLoop += m_colorLoop < halfLength ? 1 : -((m_colorLoop * 2) + 1);
+            SetSpecificLED(m_loopThroughIndividualLEDs, color);
+            SetSpecificLED(loopout, color);
+
+            m_timer = 0;
+        }
+        m_timer++;
     }
 }
+
+void DragonLeds::DiagnosticPattern(frc::DriverStation::Alliance alliance, bool coralInSensor, bool coralOutSensor, bool algaeSensor, bool intakesensor, bool questStatus, bool ll1Status, bool ll2Status, bool pigeonfaults)
+{
+    auto allianceColor = alliance == frc::DriverStation::Alliance::kBlue ? frc::Color::kBlue : frc::Color::kRed;
+    SetSpecificLED(m_diagnosticLED0, allianceColor);
+
+    auto coralInSensorcolor = coralInSensor ? frc::Color::kYellow : frc::Color::kBlack;
+    SetSpecificLED(m_diagnosticLED1, coralInSensorcolor);
```

Comment:
it would be really nice if the m_diagnosticLEDx could be descriptive 

for example, if this one were call m_coralInSensorDiagnosticLED

Reply:
Done